# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1311006561 -7200
#      Mon Jul 18 18:29:21 2011 +0200
# Node ID bc87c8da3efba3c7e229007b0eb16f8e269dd443
# Parent  9d7b0043fa4b17696849247d99b54b986fcae152
< and > get replaced, so it is no longer possible to use HTML tags in the IRC client.

diff --git a/gui/src/IRCWidget.cpp b/gui/src/IRCWidget.cpp
--- a/gui/src/IRCWidget.cpp
+++ b/gui/src/IRCWidget.cpp
@@ -155,16 +155,18 @@ IRCWidget::loginSuccessful (const char *
 }
 
 void
 IRCWidget::showPrivateMessage (const char *nick, const char *destination,
 			       const char *message)
 {
   Q_UNUSED (destination);
   QString msg (message);
+  msg.replace ("<", "&lt;");
+  msg.replace (">", "&gt;");
   if (msg.contains (m_ircClient->nickInUse ()))
     {
       msg =
 	QString ("<font color=\"#990000\"><b>%1:</b> %2</font>").arg (nick).
 	arg (msg);
     }
   else
     {
@@ -184,18 +186,21 @@ IRCWidget::showNotice (const char *nick,
   m_chatWindow->append (QString ("<font color=\"#007700\">%1</font>").
 			arg (message));
 }
 
 void
 IRCWidget::showTopic (const char *nick, const char *channel,
 		      const char *message)
 {
+  QString msg (message);
+  msg.replace ("<", "&lt;");
+  msg.replace (">", "&gt;");
   m_chatWindow->append (QString ("Topic for %2 was set by %1: %3").arg (nick).
-			arg (channel).arg (message));
+                        arg (channel).arg (msg));
 }
 
 void
 IRCWidget::showNickChange (const char *oldNick, const char *newNick)
 {
   m_chatWindow->append (QString ("%1 is now known as %2.").arg (oldNick).
 			arg (newNick));
   m_nickList.removeAll (QString (oldNick));
@@ -270,18 +275,20 @@ IRCWidget::sendMessage (QString message)
 	  m_ircClient->sendCommand (2, COMMAND_PRIVMSG,
 				    recipient.toStdString ().c_str (),
 				    pmsg.toStdString ().c_str ());
 	}
     }
   else
     {
       m_ircClient->sendPublicMessage (message);
+      message.replace ("<", "&lt;");
+      message.replace (">", "&gt;");
       m_chatWindow->append (QString ("<b>%1:</b> %2").
-			    arg (m_ircClient->nickInUse ()).arg (message));
+                            arg (m_ircClient->nickInUse ()).arg (message));
     }
 }
 
 void
 IRCWidget::sendInputLine ()
 {
   sendMessage (m_inputLine->text ());
   m_inputLine->setText ("");
diff --git a/gui/src/qirc/IRCClient.cpp b/gui/src/qirc/IRCClient.cpp
--- a/gui/src/qirc/IRCClient.cpp
+++ b/gui/src/qirc/IRCClient.cpp
@@ -279,17 +279,17 @@ IRCClient::joinChannel (QString channel)
   sendCommand (1, COMMAND_JOIN, channel.toStdString ().c_str ());
   m_recentChannel = channel;
 }
 
 void
 IRCClient::sendPublicMessage (QString message)
 {
   sendCommand (2, COMMAND_PRIVMSG, m_recentChannel.toStdString ().c_str (),
-	       message.toStdString ().c_str ());
+               message.toStdString ().c_str ());
 }
 
 void
 IRCClient::connectToServer (const char *server,
 			    int puerto,
 			    const char *nick1,
 			    const char *nick2,
 			    const char *user,
